textinput/textinput.go

```diff
@@ -180,17 +188,30 @@ func New() Model {
 var NewModel = New
 
 // SetValue sets the value of the text input.
-func (m *Model) SetValue(s string) {
+// Returns an error if validation fails.
+func (m *Model) SetValue(s string) error {
+	if m.Validate != nil {
+		err := m.Validate(s)
+		if err != nil {
+			m.Err = err
+			return err
+		}
```

Comment:
nickpicking, I would do:

```suggestion
		if err := m.Validate(s); err != nil {
			m.Err = err
			return err
		}
```